web/docs/develop-vast/write-a-plugin.md

```diff
@@ -0,0 +1,256 @@
+# Write a plugin
+
+Implementing a new plugin requires the following steps:
+
+1. [Setup the scaffolding](#setup-the-scaffolding)
+2. [Choose a plugin type](#choose-a-plugin-type)
+3. [Implement the plugin interface](#implement-the-plugin-interface)
+4. [Process configuration options](#process-configuration-options)
+5. [Compile the source code](#compile-the-source-code)
+6. [Add unit and integration tests](#add-unit-and-integration-tests)
+7. [Package it](#package-it)
+
+Next, we'll discuss each step in more detail.
+
+## Setup the scaffolding
+
+The scaffolding of a plugin includes the CMake glue that makes it possible to
+use as static or dynamic plugin.
+
+Pass `-DVAST_ENABLE_STATIC_PLUGINS:BOOL=ON` to `cmake` to build plugins
+alongside VAST as static plugins. This option is always on for static binary
+builds.
+
+VAST ships with an example plugin that showcases how a typical scaffold looks
+like. Have a look at the the [example
+plugins](https://github.com/tenzir/vast/tree/master/examples/plugins) directory,
+and an [example `CMakeLists.txt` file for
+plugins](https://github.com/tenzir/vast/blob/master/examples/plugins/analyzer/CMakeLists.txt).
+
+We recommend calling the provided `VASTRegisterPlugin` CMake in your plugin's
+`CMakeLists.txt` file instead of handrolling your CMake build scaffolding
+code. This ensures that your plugin always uses the recommended defaults.
+Non-static installations of VAST contain the `VASTRegisterPlugin.cmake` modules.
+
+If present in the plugin source directory, the build scaffolding automatically
+installs `<plugin>.yaml.example` files, commenting out every line so the file
```

Comment:
It also installs README.md and CHANGELOG.md files. Should we mention that?

Reply:
Yep.